pkg/kcp/provider/aws/vpcpeering/deleteVpcPeering.go

```diff
@@ -15,18 +18,19 @@ func deleteVpcPeering(ctx context.Context, st composed.State) (error, context.Co
 		return nil, nil
 	}
 
+	if !awsutil.CanDeletePeering(state.vpcPeering) {
```

Comment:
`if can not delete continue else delete it` - If it can not be deleted will it become orphan? 
How about to change the func name to IsAlreadyDeleted or IsBeingDeleted so the semantics becomes - `if being deleted continue else delete it`?